var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -92,11 +112,42 @@ def cmake_args(self):
                         format(spec['hipsparse'].prefix))
             args.append('-DHIPBLAS_PATH={0}'.
                         format(spec['hipblas'].prefix))
+            args.append('-DHIPRAND_PATH={0}/hiprand'.
+                        format(spec['rocrand'].prefix))
+            args.append('-DROCRAND_PATH={0}/rocrand'.
+                        format(spec['rocrand'].prefix))
+            archs = self.spec.variants['amdgpu_target'].value
+            if archs != 'none':
+                arch_str = ";".join(archs)
+                args.append(
+                    '-DGINKGO_HIP_AMDGPU={0}'.format(arch_str)
+                )
         return args
 
     @run_after('install')
-    @on_package_attributes(run_tests=True)
-    def test_install(self):
-        """Perform smoke tests on the installed package."""
+    def setup_build_tests(self):
+        """Build and install the smoke tests."""
+        # For now only develop and next releases support this scheme.
+        if not self.spec.satisfies('@develop') and not self.spec.satisfies('@1.4.0:'):
+            return
         with working_dir(self.build_directory):
             make("test_install")
+        smoke_test_path = join_path(self.build_directory, 'test_install')
+        with working_dir(smoke_test_path):
+            make("install")
+
+    def test(self):
+        """Run the smoke tests."""
+        # For now only develop and next releases support this scheme.
+        if not self.spec.satisfies('@develop') and not self.spec.satisfies('@1.4.0:'):
```

Comment:
Please output that smoke tests are being skipped.  Perhaps include a reason why.  This will help people/facilities with debugging.

Reply:
Thanks, that's a great idea, I will add that!